app/code/core/Mage/ProductAlert/Model/Observer.php

```diff
@@ -119,7 +119,7 @@ protected function _processPrice(Mage_ProductAlert_Model_Email $email)
                         if ($previousCustomer) {
                             $email->send();
                         }
-                        if (!$customer) {
+                        if (!$customer || !$customer->getId()) {
```

Comment:
I think the original `if (!$customer) {` is always `false` because ` Mage::getModel('customer/customer')->load($alert->getCustomerId());` always return the customer model, even when the ID is invalid. So, this is a bug
```suggestion
                        if (!$customer->getId()) {
```